src/components/app/details/cicdHistory/ImageTags.tsx

```diff
@@ -0,0 +1,427 @@
+                <div className="icon-dim-16 fcn-5 ml-8 cursor">
+                    <Question />
+                </div>
+            </TippyCustomized>
+        )
+    }
+
+    const getBuildContextAdditionalContent = () => {
+        return (
+            <div className="p-12 fs-13">
+                <ul>
+                    <li>
+                        A release tag can only be added if a workflow has CD pipelines deploying to Production
+                        environments.
+                    </li>
+                    <li>Multiple tags can be added to an image.</li>
+                    <li>Multiple images in an application cannot have the same tag.</li>
+                    <li>
+                        Tags cannot be deleted once saved. Although, you can soft delete a tag if an unwanted tag has
+                        been added.
+                    </li>
+                </ul>
+            </div>
+        )
+    }
+
+    const creatableRef = useRef(null)
+
+    if (newDescription === '' && displayedTags.length === 0 && !isEditing && !tagsEditable) {
+        return <div></div>
+    }
+
+    if (newDescription === '' && displayedTags.length === 0 && !isEditing) {
+        return (
+            <div className="bcn-0 pt-12 pr-12">
+                <AddImageButton handleEditClick={handleEditClick} />
+            </div>
+        )
+    }
+
+    return (
+        <div className="pt-12 pr-12">
+            {isEditing && tagsEditable ? (
+                <div className="bcn-0 dc__border-top-n1 ">
+                    <div className="cn-7 mt-12 flex left">
+                        <span>Release tags (eg. v1.0)</span>
+                        <div className="flex row ml-0">{renderInfoCard()}</div>
+                    </div>
+                    <div className="mt-6" data-testid="add-tag-text-area">
+                        <Creatable
+                            placeholder="Type a tag and press enter"
+                            onCreateOption={handleTagCreate}
+                            ref={creatableRef}
+                            components={CreatableComponents}
+                        />
+                    </div>
+
+                    {tagErrorMessage && (
+                        <div className="flex left">
+                            <Error className="form__icon form__icon--error" />
+                            <div className="form__error">{tagErrorMessage}</div>
+                        </div>
+                    )}
+                    <div className="flex-wrap mt-8 flex left">
+                        {displayedTags?.map((tag, index) => (
+                            <ImageTagButton
+                                key={tag.id}
+                                text={tag?.tagName}
+                                isSoftDeleted={tag?.deleted}
+                                isEditing={isEditing}
+                                onSoftDeleteClick={(e) => {
+                                    stopPropagation(e)
```

Comment:
Please avoid inline function

Reply:
this inline function cannot be avoided since i need event e and index as arguments .